core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -172,15 +176,51 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     }
   }
 
+  def getBillingProject(projectName: RawlsBillingProjectName): Future[PerRequestMessage] = {
+    for {
+      resourceRoles <- samDAO.listUserRolesForResource(SamResourceTypeNames.billingProject, projectName.value, userInfo)
+      maybeBillingProject <- dataSource.inTransaction { dataAccess => dataAccess.rawlsBillingProjectQuery.load(projectName) }
+    } yield {
+      maybeProjectResponse(resourceRoles, maybeBillingProject) match {
+        case Some(projectResponse) => RequestComplete(StatusCodes.OK, projectResponse)
+        case None => RequestComplete(StatusCodes.NotFound)
+      }
+    }
+  }
+
+  def listBillingProjectsV2(): Future[PerRequestMessage] = {
+    for {
+      samUserResources <- samDAO.listUserResources(SamResourceTypeNames.billingProject, userInfo)
+      projectsInDB <- dataSource.inTransaction { dataAccess => dataAccess.rawlsBillingProjectQuery.getBillingProjects(samUserResources.map(resource => RawlsBillingProjectName(resource.resourceId)).toSet) }
+    } yield {
+      val projectsByName = projectsInDB.map(p => p.projectName -> p).toMap
+      val projectResponses = samUserResources.flatMap { samUserResource =>
+        val allSamRoles = samUserResource.direct.roles ++ samUserResource.inherited.roles
```

Comment:
Should we be including any public roles as well? I doubt we have any public billing projects so likely a non issue

Reply:
public roles will show up in direct and inherited as well, the sets are not exclusive